test/bootstrap_fields_test.rb

```diff
@@ -72,7 +75,10 @@ class BootstrapFieldsTest < ActionView::TestCase
       <input name="utf8" type="hidden" value="&#x2713;"/>
       <div class="form-group">
         <label for="user_misc">Misc</label>
-        <input class="form-control-file is-invalid" id="user_misc" name="user[misc]" type="file" />
+        <div class="custom-file">
+          <input class="custom-file-input is-invalid" id="user_misc" name="user[misc]" type="file" />
+          <label class="custom-file-label" for="user_misc">Choose file</label>
+        </div>
         <div class="invalid-feedback">error for test</div>
       </div>
     </form>
```

Comment:
I would add a `form_with` test under Rails 5.1 passing in a explicit ID, and making sure the ID is on the `input` and used as the `for` attribute of the `label`.

Reply:
Good suggestion. I will give that a shot!